### ISSUE TITLE
Help

### ISSUE THREAD

**Issue by esQmo** [lykmapipo/android-push#1]
Hey, how can I base my project on this? I'm basically trying to build a push notification service backed by Firebase Messaging service so I thiught your lib would help. can you help me understand how it wok?

**Comment by lykmapipo** [comment]
@esQmo Please take a look on [sample android app](https://github.com/lykmapipo/android-push/tree/master/app) and [sample node api](https://github.com/lykmapipo/android-push/tree/master/api).

**Comment by esQmo** [comment]
Hey. I have cloned the project. I can read the code. But what I woke like to know is how it does work.

**Comment by lykmapipo** [comment]
@esQmo 

**Guides on Firebase FCM**
1. [FCM Architectural Overview](https://firebase.google.com/docs/cloud-messaging/fcm-architecture)
2. [Firebase Cloud Messaging client app on Android](https://firebase.google.com/docs/cloud-messaging/android/client)

**Library: How it works**
1. Setup firebase fcm for android clients with implementation of required services
2. Provide synchonization mechanism for android client and your api service to synchronize [common data](https://github.com/lykmapipo/android-push#android-api)

**TLDR;**
This library provide implementation of firebase fcm for android clients with extra synchronization 
mechanisms to your api endpoint.
